smack-experimental/src/main/java/org/jivesoftware/smackx/mam/provider/MamResultProvider.java

```diff
@@ -43,6 +45,7 @@
     @Override
     public MamResultExtension parse(XmlPullParser parser, int initialDepth, XmlEnvironment xmlEnvironment)
                     throws XmlPullParserException, IOException, SmackParsingException, ParseException {
+        MamElementFactory elementFactory = MamVersion.fromNamespace(parser.getNamespace()).newElementFactory();
```

Comment:
It looks like `MamVersion.fromNamespace(parser.getNamespace()).newElementFactory()` is a repeated pattern, which suggests what we may want to create a static convenience method like `MamElementFactory.for(XmlPullParser)`.

Reply:
Good idea, changed.